src/c++/perf_analyzer/genai-perf/genai_perf/export_data/report_output.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from argparse import Namespace
+
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.export_data.data_exporter_factory import (
+    DataExporterFactory,
+    DataExporterType,
+)
+from genai_perf.export_data.data_exporter_interface import DataExporterInterface
+from genai_perf.llm_metrics import LLMProfileDataParser
+from genai_perf.parser import get_extra_inputs_as_dict
+
+
+class ReportOutput:
+    """
+    A class to orchestrate output generation.
+    """
+
+    def __init__(self, data_parser: LLMProfileDataParser, args: Namespace):
+        self.data_parser = data_parser
```

Comment:
I don't think we need to set data_parser as an attribute since we never use it outside of `__init__`
```suggestion
```

Also `ReportOutput` class shouldn't really need to take `data_parser` as an argument. All it needs is just the `stats` object so we can just pass that information only.

Reply:
> Not against it persay, but I am not convinced yet.

Haha yeah of course. And you have a good point.

The reason why I said we only need `stats` is because I generally prefer minimal information to be passed around between the objects. I like this because it could potentially help us decouple the two objects which can make extending or modifying those objects later much easier.

Reply:
> I like this because it could potentially help us decouple the two objects which can make extending or modifying those objects later much easier.

Man, such great points. We are deciding between getting the stats here or in the main report_output method.
I have never had a good sense of where to draw the line so lacking that,  I always lean towards a clean high level method and putting the details in the class designed for that purpose.
My logic being, main should be the glue holding the classes together. I dont want it doing much beyond calling classes to handle the real work. That feels easy to read and understand to me.

Your approach means we have a bit more mess in the main loop but a tighter class that is more decoupled. 
The one down side I foresee is what does it look like if we have to handle multiple concurrencies/request rates. My approach, the update is hidden in this class only. Your approach, the main report_method changes and the interface to the class changes. That felt less clean to me.

Reply:
In case of multiple experiments (concurrency/req rate), I think the stats object should include those information.

Reply:
Changed the interface to take a stats object.
We will deal with the multiple experiments when we get there at another time.
Thanks for the feedback and help here.